UniqueArrayImp.h

```diff
@@ -0,0 +1,116 @@
+#ifndef MTMPARKINGLOT_UNIQUEARRAYIMP_H
+#define MTMPARKINGLOT_UNIQUEARRAYIMP_H
+
+template <class Element, class Compare>
+UniqueArray<Element, Compare>::UniqueArray(unsigned int size) :
+    data(new Element*[size]()), size(size) {}
+
+template <class Element, class Compare>
+UniqueArray<Element, Compare>::UniqueArray(const UniqueArray& other) :
+    data(new Element*[other.size]()), size(other.size)
+{
```

Comment:
according to course convientions, left brackets should not be in their own line. 
foo (args){
...
}

Reply:
Oh, good catch. Any idea what to do when the arguments are too long and don't fit in 80 chars, like here? It looks wrong no matter what I do.

Reply:
just a newline I think, it might not be the prettiest but it follows conventions.

Reply:
so, which one?
```CPP
int option1(
    LongTypeName arg )
{
    /* code */
}

int option2(
    LongTypeName arg ) {
    /* code */
}

int option3(
    LongTypeName arg
){
    /* code */
}
```
or something else?

Reply:
option2